core/ide/src/main/java/org/alice/ide/cascade/fillerinners/ConstantsOwningFillerInner.java

```diff
@@ -83,12 +82,16 @@ private ConstantsOwningFillerInner(AbstractType<?, ?, ?> type) {
 
   @Override
   public void appendItems(List<CascadeBlankChild> items, ValueDetails<?> details, boolean isTop, Expression prevExpression) {
+    this.appendItems(items, details, isTop, prevExpression, null);
+  }
+
+  public void appendItems(List<CascadeBlankChild> items, ValueDetails<?> details, boolean isTop, Expression prevExpression, Object defaultValue) {
     AbstractType<?, ?, ?> type = this.getType();
     for (AbstractField field : type.getDeclaredFields()) {
       if (field.isPublicAccess() && field.isStatic() && field.isFinal()) {
-        //todo: should this be identical? to?
+        boolean isDefault = defaultValue != null && field.getName().equals(defaultValue.toString());
```

Comment:
Could be computed inside the next `if` only when needed.
Then the two conditionals could collapse.